frontend/src/app/activity/activity-log/activity-log.component.ts

```diff
@@ -26,117 +24,71 @@ const activityIconMap = {
-        animate('220ms ease-out', style([{transform: 'translateY(0%)'}, {opacity: 1}, {height: '*'}]))
-      ]),
-      transition(':leave', [
-        animate('220ms ease-in', style([{transform: 'translateY(-40%)'}, {opacity: 0}, {height: 0}]))
-      ])
-    ])
-  ]
 })
 export class ActivityLogComponent implements OnInit, OnDestroy {
-
-  private static readonly PAGE_SIZE = 10;
-  @ViewChild('activityLogContainer', {static: true})
-  public activityLogContainer: ElementRef<HTMLElement>;
-  public currentlyLoading: boolean = false;
-  public panelHeight: number = 500;
-  private activityRest: ActivityRestService;
-  private currentPageIndex: number;
-  private loadedAll: boolean;
-  private mostRecentDate: Date;
   private subscriptions: Array<Subscription> = [];
-
-  constructor(rest: RestService,
-              private ws: WebSocketServiceWrapper,
-              private userService: UserService) {
-    this.activityRest = rest.activity();
-    this._results = [];
-    this.currentPageIndex = 0;
-    this.loadedAll = false;
+  private _activities: Array<Activity>;
+
+  public total = 0;
+  public pageSettings = {
+    pageSize: 10,
+    pageSizeOptions: [10, 25, 50, 100],
+    pageIndex: 0
+  };
+  public pendingActivities: Array<Activity> = [];
+
+  constructor(private rest: RestService,
+              private ws: WebSocketServiceWrapper) {
+    this._activities = [];
   }
 
-  private _results: Array<Activity>;
-
-  public get results(): Array<Activity> {
-    return this._results;
+  public get activities(): Array<Activity> {
+    return this._activities;
   }
 
-  public get infiniteScrollDisabled(): boolean {
-    return this.loadedAll || this.currentlyLoading;
+  public get hasPendingActivities(): boolean {
+    return this.pendingActivities.length > 0;
   }
 
   public getIcon(activity: Activity): string {
     return (activityIconMap[activity.triggerType] || activityIconMap.unknown).icon;
   }
 
   ngOnInit(): void {
-    this.userService.currentUser().subscribe(() => {
-      this.internallyLoadMore(this.determineNumberOfInitialChargesToLoad());
-      this.ws.getActivityStream().subscribe(activity => {
-        this._results.unshift(activity);
-        this.currentPageIndex = Math.floor(this.results.length / ActivityLogComponent.PAGE_SIZE);
-      });
-    });
+    this.loadStartPage();
+    this.subscriptions.push(this.ws.getActivityStream().subscribe(activity => {
+      this.pendingActivities.push(activity);
+    }));
   }
 
   ngOnDestroy() {
     this.subscriptions.forEach(subscription => subscription.unsubscribe());
   }
 
-  // the infinite scroll container needs a discrete height value
-  @HostListener('window:resize', ['$event'])
-  onResize() {
-    this.panelHeight = this.getHeight();
-  }
-
-  public loadMore() {
-    if (this.infiniteScrollDisabled) {
-      return;
+  public paginationEvent(page: PageEvent) {
+    this.loadPage(page);
+    if (page.pageIndex === 0) {
+      this.pendingActivities = [];
     }
```

Comment:
The page size is not saved in `this.pageSettings.pageSize`. If you use a different page size than the default, e.g. 100, and hit the refresh button, it will only reload 10 items.